Weather App Moa

mhammarback opened the pull request:
**Link to my deployed project:**
https://sunnydaysplease.netlify.app/

**My reflections on how this week's project turned out:**
I am quite happy with this weeks result. I did however after the demos on friday get new ideas and inputs in how to make my own project better. I decided not to do those changes - yet. But if change request is needed i will of course go back. I just did not have the time this weekend. If changes are made i would probably be able to make my code shorter and i understand that that is always good.  It was very fun working with the API, it feels "alive" so all in all a great week.


**Things I'd like to have clarified or explained in more detail:**
I think this week was pretty clear, it's more that i feel like there is so many ways to get the same result so sometimes it's hard to navigate. But i guess that will make more and more sense with more experience

jespa374 commented on index.html:
```diff
@@ -0,0 +1,72 @@
+
+<body>
+
+    <main class="weatherApp" id="weatherAppId">
+
+        <section class="todaysWeather" id="todaysWeather">
+            
+            <div id="todaysWeatherIcon"></div>
+            <div class="temperature" id="temperature"></div>
+            <div class="cityName" id="name"></div>
```
I think you should write class names and ID names in kebabcase in HTML & CSS

jespa374 commented on index.html:
```diff
@@ -0,0 +1,72 @@
+            </div>    
+        </section>
+
+            <!--Weather Forecast Stockholm-->
+        <section class="fiveDaysWeather" id="forecast">
+            <div id="forecastDay"></div>
+            <div id="forecastTemp"></div>
+            <div id="forecastIcon"></div>
+        </section>
+
+
+        <!--Todays Weather in different cities-->
```
Great that you have comments which makes it easier for the reader :)

jespa374 commented on script.js:
```diff
@@ -0,0 +1,218 @@
+
+//API KEY 
+const apiKey = '54a820a4b63e82050a15212c06998bb0';
+
+//API Stockholm 
+const apiUrl = `https://api.openweathermap.org/data/2.5/weather?q=Stockholm,Sweden&units=metric&APPID=${apiKey}`;
+const apiUrlForecast = `https://api.openweathermap.org/data/2.5/forecast?q=Stockholm,Sweden&units=metric&APPID=${apiKey}`;
```
Good that you assign the API url and the API key to variables. Looks really clean this way and very organized :)One thing is that I think the convention is to write URLs like API_URL instead of camelcase

jespa374 commented on script.js:
```diff
@@ -0,0 +1,218 @@
+const apiUrlForecastBerlin= `https://api.openweathermap.org/data/2.5/forecast?q=Berlin,Germany&units=metric&APPID=${apiKey}`;
+
+
+const nameContainer = document.getElementById('name');
+const tempContainer = document.getElementById('temperature');
+const sortContainer = document.getElementById('sort');
+const sunriseContainer = document.getElementById('sunrise');
+const sunsetContainer = document.getElementById('sunset');
+const forecastContainer = document.getElementById('forecast')
```
Really good that you assign your document.getelementids to variables :)

jespa374 commented on script.js:
```diff
@@ -0,0 +1,218 @@
+//Todays weather icon
+        document.getElementById('todaysWeatherIcon').innerHTML = `<img src=https://openweathermap.org/img/wn/${json.weather[0].icon}@2x.png></img>` 
+    })
+    .catch((error) => {
```
Good that you have .catch(error) to see if there's an error :)

EmOlsz replied:
Very good :)

jespa374 commented on script.js:
```diff
@@ -0,0 +1,218 @@
+    })
+    .catch((error) => {
+        console.log(error) 
+
```
some unnecessary space here

jespa374 commented on script.js:
```diff
@@ -0,0 +1,218 @@
+            const dayTemp = item.main.temp;
+            const weekTemp = dayTemp.toFixed(0.1);
+
+            document.getElementById('forecastDay').innerHTML += `<p>${dayName}</p>`
+            document.getElementById('forecastTemp').innerHTML += `<p>${weekTemp}°C</p>`
+            document.getElementById('forecastIcon').innerHTML += `<img src=http://openweathermap.org/img/wn/${item.weather[0].icon}@2x.png></img>`
```
Here you could have assigned these document.getelementbyids to variables

jespa374 commented on script.js:
```diff
@@ -0,0 +1,218 @@
+        return Response.json()
+    })
+    .then ((json) => {
+        console.log(json)
```
You can remove the console.log

jespa374 commented on script.js:
```diff
@@ -0,0 +1,218 @@
+        return Response.json()
+    })
+    .then ((json) => {
+        console.log(json)
```
Remove the console.log

jespa374 commented on style.css:
```diff
@@ -0,0 +1,166 @@
+
+html {
+    font-family:'Raleway', Sans-Serif;
```
Good that you have a fallback font here in case the first doesn't work :)

jespa374 commented on style.css:
```diff
@@ -0,0 +1,166 @@
+    margin: 0;
+    background-color: #3282b8;   
+}
+
+
```
Unnecessary spacing or inconsistent with the line of spacing

EmOlsz replied:
Exactly

jespa374 commented on style.css:
```diff
@@ -0,0 +1,166 @@
+}
+
+
+#todaysWeatherIcon {
+    display: flex;
+    flex-direction: row-reverse;   
+}
+
+#todaysWeatherIcon img {
```
You should avoid using IDs in styling. Use class instead

EmOlsz replied:
Agree

jespa374 commented on style.css:
```diff
@@ -0,0 +1,166 @@
+    width: 150px;
+}
+
+.temperature{
```
Should be space before the {

EmOlsz replied:
Agree

jespa374 commented on style.css:
```diff
@@ -0,0 +1,166 @@
+}
+
+
+#forecastIcon {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+}
+
+#forecastIcon img {
+    width: 35px;  
+}
+
+
+/*FAVOURITE CITY DESIGN*/
+
+#favouriteCityAll {
```
Use class

jespa374 commented on style.css:
```diff
@@ -0,0 +1,166 @@
+    justify-content: space-between;
+}
+
+#todaysWeatherIconOslo, #todaysWeatherIconOAlicante , #todaysWeatherIconBerlin {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+}
+
+#todaysWeatherIconOslo img, #todaysWeatherIconAlicante img , #todaysWeatherIconBerlin img{
```
Maybe you could add one class and apply the same class to  all of these elements

jespa374 commented on style.css:
```diff
@@ -0,0 +1,166 @@
+
+/***  Media Quaries Desktop ***/
+  
+@media screen and (min-width: 1024px) {
```
The min-width here should be 1025px since you have max-width 1024 px in the previous media query.

jespa374 reviewed (approved):
Great design! Simple and clean! Your code is really clean, well-structured and easy to read and follow with your comments. Great job! You have achieved both blue and red level goals! :) You can do the changes that I have proposed when you have the time but otherwise job well done! :)

EmOlsz commented on script.js:
```diff
@@ -0,0 +1,218 @@
+//NAME OF CITY, TEMPERATURE, SORT OF WEATHER, SUNRISE, SUNSET
+
+fetch(apiUrl)
+    .then((Response) => { 
```
We shouldn't write `Response` in capital letters - right now, during React week, I hope you understand why.
Moreover, you don't need to wrap this argument in one extra parentheses, we don't have to do that if we have only one argument

EmOlsz commented on script.js:
```diff
@@ -0,0 +1,218 @@
+        return Response.json()
+    })
+    .then ((json) => {
+        console.log(json)
```
Before final push, we should delete all console.logs

EmOlsz commented on script.js:
```diff
@@ -0,0 +1,218 @@
+        document.getElementById('todaysWeatherIcon').innerHTML = `<img src=https://openweathermap.org/img/wn/${json.weather[0].icon}@2x.png></img>` 
+    })
+    .catch((error) => {
+        console.log(error) 
```
One tip for you - we have special `console` for errors, it's called `console.error()`. It does exactly same thing as normal `console.log()`, but has red background

EmOlsz commented on script.js:
```diff
@@ -0,0 +1,218 @@
+    })
+    .then ((json) => {
+        console.log(json)
+        const filteredForecast = json.list.filter(item =>
```
Wrong indentation

EmOlsz commented on script.js:
```diff
@@ -0,0 +1,218 @@
+ const nameContainerAlicante = document.getElementById('nameAlicante');
+ const tempContainerAlicante = document.getElementById('temperatureAlicante');
+ const forecastContainerAlicante = document.getElementById('forecastAlicante');
+ const AlicanteIcon = document.getElementById('todaysWeatherIconAlicante');
```
once again, capital letter

EmOlsz reviewed (changes requested):
Good job Moa! Your code is pretty consequent and without any crucial errors. Your CSS styles are well done, same goes for HTML.
Still, some things needs to be fixed - take a look on my comments and provide changes

mhammarback commented:
Thank you both for your code reviews! 
I have implemented your requested changes here is a summarize: 
- Indentation fixed 
- Capital R in response changed to small r 
- Capital A changed to small a 
- No CSS styling on ID, changed to class
-  catch error have console.error()
